uaa/src/test/java/org/cloudfoundry/identity/uaa/mock/saml/SamlAuthenticationMockMvcTests.java

```diff
@@ -99,12 +141,192 @@ void installTestLogger() {
+        assertThat("RelayState is missing", parameterMap.get("RelayState"), notNullValue());
+        assertThat(parameterMap.get("RelayState")[0], equalTo("testsaml-redirect-binding"));
+
+        // Decode & Inflate the SAMLRequest and check the AssertionConsumerServiceURL
+        String samlRequestXml = samlDecodeAndInflate(parameterMap.get("SAMLRequest")[0]);
+        assertThat(samlRequestXml)
+                .contains("<saml2p:AuthnRequest");
+
+        XmlAssert xmlAssert = XmlAssert.assertThat(samlRequestXml)
+                .withNamespaceContext(xmlNamespaces());
+        xmlAssert.valueByXPath("//saml2p:AuthnRequest/@AssertionConsumerServiceURL")
+                .isEqualTo("http://localhost:8080/uaa/saml/SSO/alias/integration-saml-entity-id");
+        xmlAssert.valueByXPath("//saml2p:AuthnRequest/saml2:Issuer")
+                .isEqualTo("integration-saml-entity-id"); // matches login.entityID
+    }
+
+    @Test
+    void sendAuthnRequestToIdpPostBindingMode() throws Exception {
+        final String samlRequestMatch = "name=\"SAMLRequest\" value=\"";
+
+        MvcResult mvcResult = mockMvc.perform(
+                        get("/uaa/saml2/authenticate/%s".formatted("testsaml-post-binding"))
+                                .contextPath("/uaa")
+                                .header(HOST, "localhost:8080")
+                )
+                .andDo(print())
+                .andExpectAll(
+                        status().isOk(),
+                        content().string(containsString("name=\"SAMLRequest\"")),
+                        content().string(containsString("name=\"RelayState\" value=\"testsaml-post-binding\"")))
+                .andReturn();
+
+        // Decode the SAMLRequest and check the AssertionConsumerServiceURL
+        String contentHtml = mvcResult.getResponse().getContentAsString();
+        contentHtml = contentHtml.substring(contentHtml.indexOf(samlRequestMatch) + samlRequestMatch.length());
+        contentHtml = contentHtml.substring(0, contentHtml.indexOf("\""));
+        String samlRequestXml = new String(samlDecode(contentHtml), StandardCharsets.UTF_8);
+        assertThat(samlRequestXml).contains("<saml2p:AuthnRequest");
+
+        XmlAssert xmlAssert = XmlAssert.assertThat(samlRequestXml)
+                .withNamespaceContext(xmlNamespaces());
+        xmlAssert.valueByXPath("//saml2p:AuthnRequest/@AssertionConsumerServiceURL")
+                .isEqualTo("http://localhost:8080/uaa/saml/SSO/alias/integration-saml-entity-id");
+        xmlAssert.valueByXPath("//saml2p:AuthnRequest/saml2:Issuer")
+                .isEqualTo("integration-saml-entity-id"); // matches login.entityID
+    }
+
+    @Test
+    void sendAuthnRequestFromNonDefaultZoneToIdpRedirectBindingMode() throws Exception {
+        // create IDP in non-default zone
+        createMockSamlIdpInSpZone("classpath:test-saml-idp-metadata-redirect-binding.xml", "testsaml-redirect-binding");
```

Comment:
we should resolve this file to a string, e.g. using helper like getResourceAsString ... found this during tests, that SAML definition contains this string but not the XML....
Would normally fail in validateSamlIdentityProviderDefinition